File: src/main/src/main/java/com/example/demo/jooq_domain/memo/MemoParam.java
package com.example.demo.jooq_domain.memo;

import lombok.*;
import org.modelmapper.ModelMapper;

@Builder
@NoArgsConstructor(staticName = "create", access = AccessLevel.PUBLIC)
@AllArgsConstructor(staticName = "create", access = AccessLevel.PUBLIC)
@Data
public class MemoParam {

    private String title;
    private String body;

    public static MemoParam mapper(Object data) {
        return new ModelMapper().map(data, MemoParam.class);
    }

}

File: src/main/src/main/kotlin/com/example/demo/domain/memo/MemoParam.kt
package com.example.demo.domain.memo

import org.valiktor.functions.hasSize

data class MemoParam(
    val title: String,
    val body: String
) {
    init {
        org.valiktor.validate(this) {
            validate(MemoParam::title).hasSize(max = 256)
            validate(MemoParam::body).hasSize(max = 65535)
        }
    }
}
